src/cargo/util/toml/mod.rs

```diff
@@ -2146,7 +2148,12 @@ fn patch(
                         dep.unused_keys(),
                         &mut manifest_ctx.warnings,
                     );
-                    dep_to_dependency(dep, name, manifest_ctx, None)
+
+                    let patch_loc = PatchLocation::Manifest(manifest_ctx.file.to_path_buf());
+                    dep_to_dependency(dep, name, manifest_ctx, None).map(|dep| Patch {
+                        dep,
+                        loc: patch_loc,
+                    })
```

Comment:
nit: while this isn't a misuse of `.map`, I feel like being imperative here would be clearer

```rust
let dep = dep_to_dependency(dep, name, manifest_ctx, None)?;
let loc = PatchLocation::Manifest(manifest_ctx.file.to_path_buf());
Ok(Patch {
    dep,
    loc,
})
```